### PR TITLE
Feature/boilerplate

### PR DESCRIPTION
**Pull request by Wanja-Kayago**
# Description

Please include a summary of the change and which issue is fixed. 
Please also include relevant motivation and context. 
List any dependencies that are required for this change.

Perhaps answering the following questions will help create a good description:
* What is this?
* Why this?
* How are you doing this? 

Fixes # (issue) [Provide a link to the related Jira Task]

## Type of change

Specify the kind of PR/MR

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] Optimization
- [x] Improvement
- [ ] Bug fix

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A - e.g Manually tested user creation
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Laura-Nyaaga — APPROVED** [review]

**Review by yvonneannet — APPROVED** [review]

**Review by NancyMurangira — APPROVED** [review]

**Review by Faith-yiamat — APPROVED** [review]

File: Totosteps/Totosteps/settings.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+# See https://docs.djangoproject.com/en/5.1/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'django-insecure-jp_-=swbn0$%#*b*knngv@-u%cx81^vw+j!ucf#%!0r_^-h#f#'
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
+
+ALLOWED_HOSTS = ['localhost','127.0.0.1']
+
```
Comment (Ndichu-shee) [inline Totosteps/Totosteps/settings.py:38]: Why do you have these allowed hosts

File: Totosteps/Totosteps/settings.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+# Default primary key field type
+# https://docs.djangoproject.com/en/5.1/ref/settings/#default-auto-field
+
+DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
+
+AUTH0_DOMAIN = os.environ.get("AUTH0_DOMAIN")
+AUTH0_CLIENT_ID = os.environ.get("AUTH0_CLIENT_ID")
+AUTH0_CLIENT_SECRET = os.environ.get("AUTH0_CLIENT_SECRET")
+
```
Comment (Ndichu-shee) [inline Totosteps/Totosteps/settings.py:146]: Hide these credentials

Comment (Wanja-Kayago) [reply Totosteps/Totosteps/settings.py:146]: It's the code to access the real value for the variables which are stored in .env with the actual credentials

File: Totosteps/Totosteps/settings.py (hunk 1 above)
Comment (Wanja-Kayago) [reply Totosteps/Totosteps/settings.py:38]: I have removed them

**Pull request closed without merging**